File: programmers/lv1/6-실패율/Solution.java
import java.util.*;
class Solution {
    public int[] solution(int N, int[] stages) {
        double[] rate = new double[N+1];
        Integer[] answer = new Integer[N];
        int[] answer_list = new int[N];
        int n = 0;
        int count = 0;
        
        for (int i = 1; i < N+1; i++) {
            for (int s : stages) {
                if (s >= i) n++;
                if (s == i) count++;
            }
            if (n == 0) rate[i] = 0;   
            else rate[i] = (double) count / n;
            n = 0;
            count = 0;
        }
        
        for (int i = 1; i < N+1; i++) {
            answer[i-1] = i;
        }
        
        Arrays.sort(answer, (a, b) -> Double.compare(rate[b], rate[a]));
        
        for (int i = 0; i < N; i++) {
            answer_list[i] = answer[i];
        }
        
        return answer_list;
    }
}

File: programmers/lv1/6-실패율/solution.py
def solution(N, stages):
    answer = []
    n = 0
    
    for i in range(1, N+1) :
        for s in stages :
            if (s >= i) :
                n += 1
        if n == 0 :
            answer.append([i, 0])
        else :
            fail_rate = stages.count(i) / n
            answer.append([i, fail_rate])
        n = 0
        
    answer.sort(key = lambda x : -x[1])
    answer_list = [x[0] for x in answer]
    
    return answer_list